### PR TITLE
OAK-9720: Update Oak 1.22 to Jackrabbit 2.20.5

### PR DESCRIPTION
**Pull request by reschke**

### CODE REVIEW

File: oak-upgrade/pom.xml [hunk 1]
```diff
@@ -134,12 +139,12 @@
     <dependency>
       <groupId>org.apache.jackrabbit</groupId>
       <artifactId>jackrabbit-core</artifactId>
-      <version>${jackrabbit.version}</version>
+      <version>${jackrabbit.test.version}</version>
```
Comment (mreutegg) [inline oak-upgrade/pom.xml]: This and the next change are not on test dependencies. I think this is confusing.

File: oak-upgrade/pom.xml [hunk 2]
```diff
@@ -31,6 +31,11 @@
   <name>Oak upgrade</name>
   <description>Tooling for upgrading Jackrabbit repositories to Oak</description>
 
+  <properties>
+    <!-- Version of Jackrabbit used for testing, see OAK-9890 -->
+    <jackrabbit.test.version>2.20.4</jackrabbit.test.version>
```
Comment (anchela) [inline oak-upgrade/pom.xml]: hi @reschke , not sure i get the description of OAK-9890 :-(
why does the test-version needs to be 2.20.4 when the regular dependency is 2.20.5?

would it make sense to fix the tests that cause the issue?

Comment (reschke) [reply oak-upgrade/pom.xml]: The issue is that you can't deploy jackrabbit-core 2.20.5 alongside oak 1.22.x, as Jackrabbit 2.20.x is on a newer version of jackrabbit-api.

The test has a jackrabbit dependency as it compares Oak's with Jackrabbit's behavior. For *that* purpose, using an older version should be fine. Or we get rid of the test, but that would be something for you, it's related to authentication.

File: oak-upgrade/pom.xml (hunk 1 above)
Comment (reschke) [reply oak-upgrade/pom.xml]: It is; I'll adjust the comment and the property name.

File: oak-upgrade/pom.xml (hunk 2 above)
Comment (anchela) [reply oak-upgrade/pom.xml]: @reschke , i see.... what test is it? afaik i was not involved with the upgrade from jackrabbit to oak. i believe tomek worked on that. but i can take a look of needed.

**Review by anchela — APPROVED** [review]

File: oak-upgrade/pom.xml (hunk 2 above)
Comment (reschke) [reply oak-upgrade/pom.xml]: it's in oak-jcr. See https://issues.apache.org/jira/browse/OAK-9889

**Pull request merged**